Remove Accordion Script

KTS915 opened the pull request:
Following PR #197, there is no longer any need for the accordion script. This PR therefore removes it from the JS folder in `wp-admin` and removes the function calling it from `template.php` in `wp-admin/includes`

I have tested this on my own localhost test install. I have also grepped to find all instances of the accordion code being used in the admin, but I have found it only in the `template.php` file.

KTS915 committed:
Update template.php

Following PR #197, there is no longer any need for the accordion script. This PR therefore removes it from the JS folder in `wp-admin` and removes the function calling it from `template.php` in `wp-admin/includes`

KTS915 committed:
Remove jQuery accordion script from those registered in script-loader.php

KTS915 committed:
Update script-loader.php

Remove more references to the accordion script

KTS915 committed:
Reverting earlier commit because it targeted the wrong 'accordion' script

Confusingly, there are two accordion scripts in the CP admin: one is part of jQuery UI and should not be removed. This commit reverts its removal.

mattyrob commented:
I've done a little testing and the Menu accordions no longer work within Customizer once the dependency is removed:

<img width="1612" alt="Screenshot 2023-09-10 at 08 43 20" src="https://github.com/ClassicPress/ClassicPress-v2/assets/1280733/743de763-0b1c-44b3-9028-089d883bfb39">

KTS915 commented:
Good catch, @mattyrob ! I never use the Customizer so never thought of that. I'll see if I can make that work using the `details` tag instead, like the regular `nav-menu`, so that we can remove this script.

KTS915 commented:
@mattyrob, I've now taken a look and the menu accordions in the Customizer are working fine for me. Indeed, looking at the `script-loader.php` file, it's clear that the `accordion` script isn't used there. The relevant lines are these:

`
$scripts->add( 'customize-nav-menus', "/wp-admin/js/customize-nav-menus$suffix.js", array( 'jquery', 'wp-backbone', 'customize-controls', 'nav-menu', 'wp-sanitize' ), false, 1 );
$scripts->add( 'customize-preview-nav-menus', "/wp-includes/js/customize-preview-nav-menus$suffix.js", array( 'jquery', 'wp-util', 'customize-preview', 'customize-selective-refresh' ), false, 1 );
`

No mention of `accordion` there. I'm wondering if you've removed the wrong accordion version for your test. (As you can see in this PR, I made that mistake myself initially, and removed the one associated with jQuery UI.)

mattyrob commented:
@KTS915 - I'm checked out on this branch and when I get to the screen that I posted a screen grab off before, in Safari and FF, I cannot get the items in the accordion under the "Add Items" to open.

By reverting the changes in commit 06e6037c45bacb3a96953ea987ba224fbb4ea07e, that section works again.

I should say that anything currently in the Menu seems to work fine.

KTS915 committed:
Deatils and summary tags added to class-wp-customize-nav-menus.php

This replicates the changes made to the regular `nav menus` screen in that it adds the `details` and `summary` tags to the Add Menus section in the Customizer.

This change both improves accessibility and enables the `accordion` script to be removed.

KTS915 committed:
Update common.css

This deletes `display: none;` from the `.accordion-section-content` class and so enables the relevant sections to become visible when toggled using the `details` tag.

KTS915 commented:
@mattyrob Thanks, I managed to replicate the problem and have now pushed some changes to make it work. Specifically, I have added the `details` and `summary` tags, as with the regular `nav menu`, and deleted one line of CSS that would prevent the content of each section from being displayed.

xxsimoxx commented:
I'm getting wrong icon in customizer.

<img width="353" alt="Schermata 2023-09-11 alle 10 19 55" src="https://github.com/ClassicPress/ClassicPress-v2/assets/29772709/5d998d4b-2600-48df-b3b7-a8515bc89814">

The problem started in #197 but maybe can be fixed in this PR.

mattyrob committed:
Coding standards and docblock corrections

mattyrob committed:
Fix for failing unit test

mattyrob commented:
@xxsimoxx - I think I may have fixed the icon issue but was working on develop and pushed it by mistake - so I hope it was right.
If not we can revert and do more testing on this PR.

xxsimoxx commented:
@mattyrob the commit in develop fixed the issue.

xxsimoxx reviewed (approved):
Looks good to me.

mattyrob commented:
Is there one more things for this is to consider on this PR, removing `src/wp-admin/js/accordion.js ` from core?

KTS915 committed:
Delete src/wp-admin/js/accordion.js

Delete file as it is no longer used

KTS915 commented:
Lol, of course I should have deleted that file! Duh! Now done.
